dominion/src/main/java/org/codice/dominion/options/Option.java

```diff
@@ -54,8 +56,43 @@
           return clazz.cast(extension);
         }
       }
+      return null;
+    } catch (VirtualMachineError | DominionException e) {
+      throw e;
+    } catch (Throwable t) {
       throw new OptionException(
-          "unable to find an option extension for: " + annotation.annotationType());
+          "failure retrieving an option extension for: " + annotation.annotationType(), t);
+    }
+  }
+
+  /**
+   * Gets all option extensions from a specific subclass for the given option annotation.
+   *
+   * <p><i>Note:</i> The implementation provided here will search registered factories for a given
+   * annotation and returns a stream of all extensions that are an instance of the expected class.
+   *
+   * @param <E> the type of extension expected
+   * @param clazz the specific class of option extensions expected
+   * @param annotation the option annotation for which to create extension points (this will be of
+   *     the annotation class annotated with {@link Annotation})
+   * @return the corresponding extensions to use or empty if none found
+   * @throws OptionException if a failure occurred while retrieving the extension
+   */
+  @SuppressWarnings("squid:S1181" /* catching VirtualMachineError first */)
+  public static <E extends Extension> List<E> getExtensions(
+      Class<E> clazz, java.lang.annotation.Annotation annotation) {
+    try {
+      final List<E> extensions = new ArrayList<>(4);
```

Comment:
:question: Why have an initial size of 4?

Reply:
because I prefer sizing it as close to as  possible to the mex of what will ever be needed. It keeps memory footprint to a minimum and in this case, there won't ever be that many more